### PR TITLE
[workers][SIMS #1346]Enabled E2E Tests

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
- Add .vscode/launch.json to git with some debug configurations. I would like to recommend that we share this to make easy to everyone have the debug handy and for new developers joining the project to take advantage of it. Right now the below options are available.
image
- Added more factories to `@sims/test-utils` (current ones from API were not removed).
- Created one E2E test to validate the service+db, receiving a payload for a reassessment and generating an overaward.
- Created the script `test:e2e:workers`, not been added to GitHub execution tests yet.
- Sonarcloud issues are not relevant right now.

### CODE REVIEW

File: sources/packages/backend/apps/workers/src/services/application/application.service.e2e.skip-spec.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ describe.skip("ApplicationService", () => {
     jest.spyOn(msfaaNumberService, "createMSFAANumber");
   });
 
-  describe("associateMSFAANumber", () => {
+  describe.skip("associateMSFAANumber", () => {
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/workers/src/services/application/application.service.e2e.skip-spec.ts:80]: This test is not "active". Just renamed it right now to be outside the worker's E2E tests execution.
It must be fixed. It was moved here during previous PRs.

File: sources/packages/backend/apps/workers/src/services/disbursement-schedule/tests/disbursement-payloads.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { DisbursementValueType } from "@sims/sims-db";
+import { getISODateOnlyString } from "@sims/utilities";
+import { DisbursementSaveModel } from "../disbursement-schedule.models";
+
+export function createFakeDisbursementPayload(): DisbursementSaveModel[] {
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/workers/src/services/disbursement-schedule/tests/disbursement-payloads.ts:5]: Why is this not created in test-utils?

File: sources/packages/backend/libs/test-utils/src/factories/student.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export function createFakeStudent(user?: User): Student {
+  const student = new Student();
+  student.user = user ?? createFakeUser();
+  student.birthDate = faker.date.past(18).toISOString();
+  student.gender = "X";
+  student.contactInfo = {
+    address: {
+      addressLine1: faker.address.streetAddress(),
+      city: faker.address.city(),
+      country: "CAN",
```
Comment (ann-aot) [inline sources/packages/backend/libs/test-utils/src/factories/student.ts:14]: isn't it?
```
      country: "canada",
      selectedCountry: "Canada",
```

Comment (andrewsignori-aot) [reply sources/packages/backend/libs/test-utils/src/factories/student.ts:14]: Probably, yes. I will update it.

File: sources/packages/backend/apps/workers/src/services/disbursement-schedule/tests/disbursement-payloads.ts (hunk 2 above)
Comment (ann-aot) [reply sources/packages/backend/apps/workers/src/services/disbursement-schedule/tests/disbursement-payloads.ts:5]: i also didn't get this

File: sources/packages/backend/libs/test-utils/src/factories/student.ts (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/libs/test-utils/src/factories/student.ts]: Changed.

File: sources/packages/backend/libs/test-utils/src/factories/location-fake.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import * as faker from "faker";
+import { Institution, InstitutionLocation } from "@sims/sims-db";
+import { createFakeInstitution } from "@sims/test-utils";
+
+export function createFakeLocation(
```
Comment (ann-aot) [inline sources/packages/backend/libs/test-utils/src/factories/location-fake.ts:5]: mm... we do have a `createFakeInstitutionLocation`, `sources/packages/backend/libs/test-utils/src/factories/institution-location.ts` right? so, do we need this?

Comment (andrewsignori-aot) [reply sources/packages/backend/libs/test-utils/src/factories/location-fake.ts:5]: Leftover, removing it.

Comment (andrewsignori-aot) [reply sources/packages/backend/libs/test-utils/src/factories/location-fake.ts]: Removed

**Review by dheepak-aot — APPROVED** [review]
LGTM 👍

**Review by ann-aot — APPROVED** [review]
Good work @andrewsignori-aot . Thanks for doing the changes 👍

**Review by andrepestana-aot — APPROVED** [review]
Looks good!

File: sources/packages/backend/apps/workers/src/services/disbursement-schedule/tests/disbursement-payloads.ts (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/workers/src/services/disbursement-schedule/tests/disbursement-payloads.ts:5]: Talked over a call, hope that it is fine now 😉

**Comment by andrewsignori-aot** [comment]
> Looks good!

Thanks for bringing up the issue with the `Canada Grant` vs `Canada Loan` good catch and a nice fix over the call 😉

**Review by guru-aot — APPROVED** [review]
Nice work @andrewsignori-aot

**Pull request merged**